passionscotch/app/controllers/index.rb

```diff
@@ -0,0 +1,112 @@
+enable :sessions
+
+get '/' do
+  redirect 'login'
+end
+
+get '/login' do
+  erb :login
+end
+
+
+post '/login' do
+  user = User.find_by(email: params[:email])
+  if user.authenticate(params[:password])
+    session[:user_id] = user.id
+    redirect "/#{user.username}"
+  else
+    @login_error = "Sorry, that didn't work. Please try again."
+    erb :login
+  end
+end
+
+post '/register' do
+  new_user = User.create(username: params[:username], email: params[:email], password: params[:password], name: params[:name])
```

Comment:
You're not checking to make sure this user was created. You either need to use `User.create!(…)` or check the return value to ensure the user was created.